wsgidav/util.py

```diff
@@ -408,8 +408,10 @@ def safe_re_encode(s, encoding_to, errors="backslashreplace"):
         encoding_to = "ASCII"
     if compat.is_bytes(s):
         s = s.decode(encoding_to, errors=errors).encode(encoding_to)
-    else:
+    elif hasattr(s, "encode"):
```

Comment:
This method should not implement permissive fallbacks. It is up to the caller to pass in valid arguments.